sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -231,6 +236,11 @@ object SerializerSupport {
  * @param dataType The expected return type of the function call
  * @param functionName The name of the method to call.
  * @param arguments An optional list of expressions to pass as arguments to the function.
+ * @param methodInputTypes An optional list of data types specifying the input types for the method
```

Comment:
Shall we use `inputTypes: Seq[AbstractDataType]` directly? Nil means no implicit cast needed.

Reply:
Oops you are right - didn't notice the `Nil` case. We can't do this for `Invoke` though since its children include the `targetObject`.